mt-kahypar/partition/refinement/fm/strategies/local_gain_cache_strategy.h

```diff
@@ -243,8 +242,7 @@ class LocalGainCacheStrategy {
       if (i != from && i != kInvalidPartition) {
         const HypernodeWeight to_weight = phg.partWeight(i);
         const HyperedgeWeight penalty = gain_cache.benefitTerm(u, i);
-        if ( ( penalty > to_benefit || ( penalty == to_benefit && to_weight < best_to_weight ) ) &&
-             to_weight + wu <= context.partition.max_part_weights[i] ) {
+        if ( ( penalty > to_benefit || (penalty == to_benefit && to_weight < best_to_weight) ) ) {
```

Comment:
Please add some documentation to the bestOfThree function, elaborating why ignoring balance is the sensible thing to do here.